src/components/LoginButton.jsx

```diff
@@ -2,12 +2,24 @@ import React, { useEffect, useState } from 'react';
 import firebase from 'firebase';
 import Button from '@material-ui/core/Button';
 import LoginDialog from './LoginDialog';
+import ProfileMenu from './ProfileDropdown';
 
 export default function LoginButton() {
   const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
   const [loginDialogOpen, setLoginDialogOpen] = useState(false); // initialized to false
   const handleClose = () => setLoginDialogOpen(false);
 
+  const { currentUser } = firebase.auth();
+  let userName;
+  let userEmail;
+  let photoUrl;
+
+  if (currentUser != null) {
+    userName = currentUser.displayName;
+    userEmail = currentUser.email;
+    photoUrl = currentUser.photoURL;
+  }
```

Comment:
Just inline these values below where you know the user is signed in already. E.g. `name={user.displayName}`

Reply:
Alright what I did was used `useState` and just changed state of `name` when firebase auth change happens to the user's name